app/models/ability.rb

```diff
@@ -0,0 +1,16 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    can :read, Post, Public: true
+    can :read, Comment, Public: true
+
+    return unless user.present?
+
+    can :read, :all
+
+    return unless user.role == 'admin'
+
+    can :manage, :all
+  end
```

Comment:
According to the [project requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_add_authorization.md), A user can delete a `post` and a `comment` if it is theirs. please use CanCanCan for this authorization.  😊

For example:
```ruby
can :manage, Post, author: user`
```